design/bytecode/README.md

```diff
@@ -1,82 +1,117 @@
 # Pact Byte Code
-Pact byte code is a DSL for expressing logical assertions on dynamic input data.  
-The assertions can be independent or linked to form more complex clauses ('and, or').  
-Pact's goal is to be an embedded DSL for constraining the use of doughnut certificates.  
-Attaching pact bytecode to a doughnut can be thought of as attaching contractual "terms of use".  
+Pact byte code is a DSL for expressing logical assertions on dynamic input data.
+The assertions can be independent or linked to form more complex clauses ('and, or').
+Pact's goal is to be an embedded DSL for constraining the use of doughnut certificates.
+Attaching pact bytecode to a doughnut can be thought of as attaching contractual "terms of use".
 It is the pact interpreters job to resolve whether the clauses were maintained or not given
-some input data.  
+some input data.
 
 ## Grammar
-The syntax of pact byte code is organized so that interpretation requires zero look-ahead.  
-To acheive this, the language's opcodes are split into 3 loose categories: loads, conjunctions, and comparators.  
-- Comparators: express a logical comparison between two things  
-- Conjunctions: express a logical join between comparators  
+The syntax of pact byte code is organized so that interpretation requires zero look-ahead.
+To acheive this, the language's opcodes are split into 3 loose categories: loads, conjunctions, and comparators.
+- Comparators: express a logical comparison between two things
+- Conjunctions: express a logical join between comparators
 - Load: Load data from a register for comparison
 
 ```
 CONTRACT: CLAUSE*
 CLAUSE: ASSERTION*
-ASSERTION: COMPARATOR LOAD LOAD | ASSERTION CONJUNCTION ASSERTION
-CONJUNCTION: AND | OR | XOR
-COMPARATOR: EQ | GT | GTE | LT | LTE | IN
-LOAD: LD_INPUT | LD_USER
+ASSERTION: COMPARATOR LOAD_INDEX LOAD_INDEX | ASSERTION CONJUNCTION ASSERTION
+CONJUNCTION: (INVERSION + CONJUNCTION_OP)
+COMPARATOR: (INVERSION + LOAD + COMPARATOR_OP)
+INVERSION: NORMAL | NOT
+CONJUNCTION_OP: AND | OR | XOR
+LOAD: LOAD_INPUT_VS_USER | LD_INPUT_VS_INPUT
+COMPARATOR_OP: EQ | GT | GTE | IN
 ```
 
-Goals:  
+Goals:
 - Embed within doughnuts (compact)
 - Executable by the pact interpreter
 
 ## OpCodes
-8-bit big endian opcode  
-  - 6 high bits: index  
-  - 2 low bits: reserved  
+8-bit big endian opcode:
 
-comparator and conjunction opcodes are stand-alone 8bit.  
-load opcodes are followed by an 8bit index of the datum to load from the target table.  
+| bits    | 7           | 6   |      5 - 2 |    1 - 0 |
+|:--------|:-----------:|:---:|:----------:|:--------:|
+| purpose | conjunction | not |  operation | RESERVED |
```

Comment:
optional: perhaps Op(erator)Type better describes the purpose than "conjunction" for bit 7 instead

Reply:
Nice, I agree 👍